IntelliJ IDEA guide: usage (#516)
windup/windup-documentation: docs/topics/intellij-idea-plugin-run-configuration.adoc

// Module included in the following assemblies:
//
// * docs/intellij-idea-plugin-guide/master.adoc

[id="intellij-idea-plugin-run-configuration_{context}"]

> = Configuring a run configuration

Author: Done. Forgot to comment earlier today.

> You can configure multiple run configurations to run against each project you import to InteeliJ IDEA.

Reviewer: InteeliJ

.Procedure

> . In *Projects* view, click the project you want to analyze.

Author: 1. "the" -- OK. I've one back and forth with using the definite article with "view" over the last decade, but I'll go with it.
2. The GUI object is named Products, so the use of bold is correct.  https://redhat-documentation.github.io/supplementary-style-guide/#user-interface-elements
3. "Activity" in "The Activity bar" is not bolded because I am using the generic term. The interface does not have labeled sections.

> . In *Projects* view, click the project you want to analyze.

Reviewer 2: RE 3, ping me on chat. I'm not sure what you mean by "generic term".

. On the left side of the screen, click the *{ProductName}* tab.
+
If this is your first configuration, the run configuration panel is displayed on the right.

. If this is not your first configuration, right-click on one of the configurations in the list and select *New configuration*.
+

> The run configuration panel is displayed on the right.

Reviewer 2: Is "run configuration" a GUI element? (Should it be bold?)

Author: The element is unnamed, so I didn't bold it. I'm following the guidance Avital gave for the VS Code guide -- both UIs have named and unnamed elements. I've bolded he named elements and left the unnnamed elements in plain text. I'd prefer to keep the style consistent over the guides. if you feel strongly it should be changed, I will have to change other guides in a separate PR. I think the convention i'm using is a good choice, but it's your call.

Reviewer 2: OK, just asking. Sounds good. I'm just wondering how the user will understand what you're talking about considering that it's unnamed. I'm guessing that it's obvious enough.

. Complete the following configuration fields:

** *mta-cli*: Click *Add*, select the filepath, and click *Save*.

> ** *Input*: Click *Add* and do one of the following:

Reviewer 2: I don't think you need to give two options here. This is pretty elementary. I think the bottom line here is that you need to add an input file or directory. It should be fairly obvious to anyone savvy enough to be reading this document that you can either enter the path manually or using **Open File Explorer**.

```suggestion
** *Input*: Click *Add* and enter the input file or directory.
```

Author: Good point. Done.

*** Enter the input file or directory and press Enter.
*** Click *Open File Explorer*, select the directory, and click *OK*.

** *Target*: Select one or more target migration paths.
+
[NOTE]
====

> The location shown in the *Output* is set by the plugin.

Reviewer 2: Passive voice.
```suggestion
The plugin sets the location shown in the *Output* field.
```

Author: The IBM Style Guide says that the passive voice is acceptable when the system performs the action (so, no change).
![passive](https://user-images.githubusercontent.com/60698649/155504764-7fd81163-abc6-4055-8688-20a8f73c716a.png)

Reviewer 2: OK, thanks for showing me that!

====

> . In the list of configurations, right-click on the new configuration and select *Run Analysis*.

Reviewer 2: ```suggestion
. In the list of configurations, right-click the new configuration and select *Run Analysis*.
```

Author: Done.

+
The *Console (MTA)* terminal emulator opens, displaying information about the progress of the analysis.
+

> When the analysis is completed, you can click either *Report* or *Results* below the name of the configuration file you ran.

Reviewer 2: ```suggestion
When the analysis is complete, you can click either *Report* or *Results* below the name of the configuration file you ran.
```
Also...you _can_? Meaning you don't have to? If you have to, or you should, then get rid of "you can".

Author: RH documentation uses the verb "can" consistently in this manner. Given that it is physically possible to do so but not obligatory, "can" and "may" are correct grammatically, but since only an authenticated authorized user _can_ reach this step. there's a case to be made for "can" (so, no change).

Reviewer 2: +1 on _can_ instead of _may_. I was just wondering if it was obligatory.

** *Reports* opens the MTA report, which describes any issues you need to address before you migrate or modernize your application. For more information, see link:{ProductDocUserGuideURL}#review-reports_cli-guide[MTA report] in the _{UserCLIBookName}_.

> ** *Results* opens a directory displaying Hints (issues) per application.

Reviewer 2: Why is Hints capitalized here?

Author: You're right. I'm not using it as a proper noun or a GUI element (and there's no need to). Changed.

// * docs/intellij-idea-plugin-guide/master.adoc

[id="intellij-idea-plugin-run-configuration_{context}"]
= Creating a run configuration

> You can configure multiple run configurations to run against each project you import to IntelliJ IDEA.

Reviewer 2: The same comment for line 6 applies here and anywhere else you have anything like "configure a run configuration"

Author: Fixed.
